arch/configure.defaults

```diff
@@ -16,7 +16,7 @@ CC              =       CONFIGURE_CC
 LD              =       $(FC)
 RWORDSIZE       =       CONFIGURE_RWORDSIZE
 PROMOTION       =       #-fdefault-real-8
-ARCH_LOCAL      =       -DNEC -DNONSTANDARD_SYSTEM_SUBR  -DWRF_USE_CLM
+ARCH_LOCAL      =       -DNEC -DNONSTANDARD_SYSTEM_SUBR  CONFIGURE_D_CTSM
```

Comment:
@negin513 
Negin,
For your `CONFIGURE_D_CTSM` flag modification, you have accidentally removed the `-DWRF_USE_CLM` cpp flag.

Reply:
Dave, I agree with you. I think we should keep the `-DWRF_USE_CLM` as it is not relevant to CTSM coupling at all. 
I think @ billsacks removed this flag while working on this and I want to confirm with him if it is okay to add `-DWRF_USE_CLM` back to `arch/configure.defaults`. @billsacks can you please confirm this? If so I will change this line to:
```
ARCH_LOCAL      =       -DNEC -DNONSTANDARD_SYSTEM_SUBR -DWRF_USE_CLM  CONFIGURE_D_CTSM
```

Reply:
@negin513 
Negin,
Yes, I agree that everywhere you have replaced `-DWRF_USE_CLM` with `CONFIGURE_D_CTSM`, we need to have BOTH of those options listed.

Reply:
@davegill - This was an intentional change that I made. The problem with having `-DWRF_USE_CLM` when we're compiling with CTSM is that there are module name collisions at link time, because many of the modules in module_sf_clm have the same name as modules in the CTSM library. So the logic here is to turn off `-DWRF_USE_CLM` if the user is building with CTSM, but to keep it on otherwise. We have a partially-complete project to rename all modules in CTSM to add a `ctsm_` prefix, which would get around this issue. Once that's done, we could have both `-DWRF_USE_CLM` and `-DWRF_USE_CTSM` at the same time, but I'm not sure what the priority is of that. If you feel it's important to keep the old CLM code compiled in the same build as CTSM, then we can bump up the priority of that rename to allow this.

Reply:
@davegill based on what @billsacks is saying, I should go ahead and remove the `-DWRF_USE_CTSM` flag. or @davegill do you think some other solution that takes care of this?